flake.nix

```diff
@@ -31,14 +31,13 @@
             || die 'Test failed'
         '';
 
-      in {
-        packages = rec {
-          default = nil;
-          nil = let
+        nil =
+          let
             mtime = self.lastModifiedDate;
             date = "${substring 0 4 mtime}-${substring 4 2 mtime}-${substring 6 2 mtime}";
             rev = self.shortRev or (throw "Git changes are not committed");
-          in pkgs.rustPlatform.buildRustPackage {
+          in
+          pkgs.rustPlatform.buildRustPackage {
```

Comment:
We should use `rustPlatform` and `nix` from `final` for the overlay, rather than locked `pkgs`. Otherwise it always pulls in multiple nixpkgs and is not quite an "overlay".